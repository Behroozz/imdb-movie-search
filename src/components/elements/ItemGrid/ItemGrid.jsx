import React from 'react'
import { get } from 'lodash'
import { connect } from 'react-redux'

import ItemTile from '../ItemTile/ItemTile.jsx'


export const ItemGrid = ({ items, term }) => {
  console.log(items)
  return (
    <React.Fragment>
      {term && <h2 className="ml-4 mt-4">Results for {term}</h2>}
      <div className="w-full flex flex-wrap mt-4 pb-4 px-2">
        
        {
          items.length > 0 && items.map(item => <ItemTile key={item.imdbID} item={item} />)
        }
        {
          items.length === 0 && <div>No results</div>
        }
      </div>
    </React.Fragment>
  )
}

const mapStateToProps = (state) => {
  return {
    items: state.data.data,
    term: state.term
  }
}

export default React.memo(connect(mapStateToProps)(ItemGrid))