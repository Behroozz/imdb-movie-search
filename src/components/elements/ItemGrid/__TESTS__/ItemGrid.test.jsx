import React from 'react'
import renderer from 'react-test-renderer'
import { ItemGrid } from '../ItemGrid.jsx'

const dataMock = [
  {
    Title: "Waterworld",
    Year: 1995,
    imdbID: "tt0114898"
  },
  {Title: "The Making of 'Waterworld'", Year: 1995, imdbID: "tt2670548"},
  {Title: "Waterworld 4: History of the Islands", Year: 1997, imdbID: "tt0161077"},
  {Title: "Waterworld", Year: 1997, imdbID: "tt0455840"},
  {Title: "Waterworld", Year: 1997, imdbID: "tt0390617"},
  {Title: "Swordquest: Waterworld", Year: 1983, imdbID: "tt2761086"},
  {Title: "Behind the Scenes of the Most Fascinating Waterworld on Earth: The Great Backwaters, Kerala.", Year: 2014, imdbID: "tt5847056"},
  {Title: "Louise's Waterworld", Year: 1997, imdbID: "tt0298417"},
  {Title: "Waterworld", Year: 2001, imdbID: "tt0381702"},
]

test('should render a grid of 9 items', () => {
  const component = renderer.create(
    <ItemGrid items={dataMock} />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})