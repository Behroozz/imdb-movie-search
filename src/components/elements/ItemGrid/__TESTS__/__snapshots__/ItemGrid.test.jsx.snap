// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a grid of 9 items 1`] = `
<div
  className="w-full flex flex-wrap mt-4 pb-4 px-2"
>
  <div
    className="mt-4 w-full md:w-1/2 lg:w-1/3 xl:w-1/5 p-2"
  >
    <div
      className="bg-white h-full py-2 px-4"
    >
      <h3
        className="text-xl"
      >
        Waterworld
      </h3>
      <h4>
        Year : 
        1995
      </h4>
      <img
        className="w-full"
        sizes=""
        srcSet="https://via.placeholder.com/400, https://via.placeholder.com/640, https://via.placeholder.com/400"
      />
      <a
        className="mt-4 text-blue-500 block"
        href="https://imdb.com/title/tt0114898"
        target="_blank"
      >
        See more information on 
        Waterworld
      </a>
    </div>
  </div>
  <div
    className="mt-4 w-full md:w-1/2 lg:w-1/3 xl:w-1/5 p-2"
  >
    <div
      className="bg-white h-full py-2 px-4"
    >
      <h3
        className="text-xl"
      >
        The Making of 'Waterworld'
      </h3>
      <h4>
        Year : 
        1995
      </h4>
      <img
        className="w-full"
        sizes=""
        srcSet="https://via.placeholder.com/400, https://via.placeholder.com/640, https://via.placeholder.com/400"
      />
      <a
        className="mt-4 text-blue-500 block"
        href="https://imdb.com/title/tt2670548"
        target="_blank"
      >
        See more information on 
        The Making of 'Waterworld'
      </a>
    </div>
  </div>
  <div
    className="mt-4 w-full md:w-1/2 lg:w-1/3 xl:w-1/5 p-2"
  >
    <div
      className="bg-white h-full py-2 px-4"
    >
      <h3
        className="text-xl"
      >
        Waterworld 4: History of the Islands
      </h3>
      <h4>
        Year : 
        1997
      </h4>
      <img
        className="w-full"
        sizes=""
        srcSet="https://via.placeholder.com/400, https://via.placeholder.com/640, https://via.placeholder.com/400"
      />
      <a
        className="mt-4 text-blue-500 block"
        href="https://imdb.com/title/tt0161077"
        target="_blank"
      >
        See more information on 
        Waterworld 4: History of the Islands
      </a>
    </div>
  </div>
  <div
    className="mt-4 w-full md:w-1/2 lg:w-1/3 xl:w-1/5 p-2"
  >
    <div
      className="bg-white h-full py-2 px-4"
    >
      <h3
        className="text-xl"
      >
        Waterworld
      </h3>
      <h4>
        Year : 
        1997
      </h4>
      <img
        className="w-full"
        sizes=""
        srcSet="https://via.placeholder.com/400, https://via.placeholder.com/640, https://via.placeholder.com/400"
      />
      <a
        className="mt-4 text-blue-500 block"
        href="https://imdb.com/title/tt0455840"
        target="_blank"
      >
        See more information on 
        Waterworld
      </a>
    </div>
  </div>
  <div
    className="mt-4 w-full md:w-1/2 lg:w-1/3 xl:w-1/5 p-2"
  >
    <div
      className="bg-white h-full py-2 px-4"
    >
      <h3
        className="text-xl"
      >
        Waterworld
      </h3>
      <h4>
        Year : 
        1997
      </h4>
      <img
        className="w-full"
        sizes=""
        srcSet="https://via.placeholder.com/400, https://via.placeholder.com/640, https://via.placeholder.com/400"
      />
      <a
        className="mt-4 text-blue-500 block"
        href="https://imdb.com/title/tt0390617"
        target="_blank"
      >
        See more information on 
        Waterworld
      </a>
    </div>
  </div>
  <div
    className="mt-4 w-full md:w-1/2 lg:w-1/3 xl:w-1/5 p-2"
  >
    <div
      className="bg-white h-full py-2 px-4"
    >
      <h3
        className="text-xl"
      >
        Swordquest: Waterworld
      </h3>
      <h4>
        Year : 
        1983
      </h4>
      <img
        className="w-full"
        sizes=""
        srcSet="https://via.placeholder.com/400, https://via.placeholder.com/640, https://via.placeholder.com/400"
      />
      <a
        className="mt-4 text-blue-500 block"
        href="https://imdb.com/title/tt2761086"
        target="_blank"
      >
        See more information on 
        Swordquest: Waterworld
      </a>
    </div>
  </div>
  <div
    className="mt-4 w-full md:w-1/2 lg:w-1/3 xl:w-1/5 p-2"
  >
    <div
      className="bg-white h-full py-2 px-4"
    >
      <h3
        className="text-xl"
      >
        Behind the Scenes of the Most Fascinating Waterworld on Earth: The Great Backwaters, Kerala.
      </h3>
      <h4>
        Year : 
        2014
      </h4>
      <img
        className="w-full"
        sizes=""
        srcSet="https://via.placeholder.com/400, https://via.placeholder.com/640, https://via.placeholder.com/400"
      />
      <a
        className="mt-4 text-blue-500 block"
        href="https://imdb.com/title/tt5847056"
        target="_blank"
      >
        See more information on 
        Behind the Scenes of the Most Fascinating Waterworld on Earth: The Great Backwaters, Kerala.
      </a>
    </div>
  </div>
  <div
    className="mt-4 w-full md:w-1/2 lg:w-1/3 xl:w-1/5 p-2"
  >
    <div
      className="bg-white h-full py-2 px-4"
    >
      <h3
        className="text-xl"
      >
        Louise's Waterworld
      </h3>
      <h4>
        Year : 
        1997
      </h4>
      <img
        className="w-full"
        sizes=""
        srcSet="https://via.placeholder.com/400, https://via.placeholder.com/640, https://via.placeholder.com/400"
      />
      <a
        className="mt-4 text-blue-500 block"
        href="https://imdb.com/title/tt0298417"
        target="_blank"
      >
        See more information on 
        Louise's Waterworld
      </a>
    </div>
  </div>
  <div
    className="mt-4 w-full md:w-1/2 lg:w-1/3 xl:w-1/5 p-2"
  >
    <div
      className="bg-white h-full py-2 px-4"
    >
      <h3
        className="text-xl"
      >
        Waterworld
      </h3>
      <h4>
        Year : 
        2001
      </h4>
      <img
        className="w-full"
        sizes=""
        srcSet="https://via.placeholder.com/400, https://via.placeholder.com/640, https://via.placeholder.com/400"
      />
      <a
        className="mt-4 text-blue-500 block"
        href="https://imdb.com/title/tt0381702"
        target="_blank"
      >
        See more information on 
        Waterworld
      </a>
    </div>
  </div>
</div>
`;
